### ISSUE TITLE
Non-static inheritance

### ISSUE THREAD

**Issue by xeno-by** [twitter/rsc#97]
```scala
object M {
  type T = Iterator[Int]
}

abstract class C extends M.T
```

```
Test.scala:5: error: compiler crash
abstract class C extends M.T
^
rsc.util.CrashException: _empty_/M.T#
	at rsc.util.CrashException$.apply(CrashException.scala:24)
	at rsc.util.ErrorUtil$class.crash(ErrorUtil.scala:18)
	at rsc.util.package$.crash(package.scala:5)
	at rsc.outline.Symtab$scopes$.apply(Symtab.scala:25)
	at rsc.semanticdb.Semanticdb$TemplateSemanticdbOps.superClass$1(Semanticdb.scala:281)
	at rsc.semanticdb.Semanticdb$TemplateSemanticdbOps.parents(Semanticdb.scala:318)
	at rsc.semanticdb.Semanticdb$OutlineSemanticdbOps.signature(Semanticdb.scala:198)
	at rsc.semanticdb.Semanticdb.apply(Semanticdb.scala:45)
	at rsc.Compiler.rsc$Compiler$$semanticdb(Compiler.scala:175)
	at rsc.Compiler$$anonfun$tasks$5.apply$mcV$sp(Compiler.scala:79)
	at rsc.Compiler$$anonfun$run$2.apply(Compiler.scala:34)
	at rsc.Compiler$$anonfun$run$2.apply(Compiler.scala:31)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at rsc.Compiler.run(Compiler.scala:31)
	at rsc.cli.Main$.run(Main.scala:21)
	at rsc.cli.Main$.main(Main.scala:11)
	at rsc.cli.Main.main(Main.scala)
```

**Comment by xeno-by** [comment]
Just like in #90, the hardest prerequisite for implementing this - unification of scoper and outliner - is already done. Also, we needed fundamental upgrades to our symbol table and @maxov delivered, unblocking this ticket from another angle.

**Comment by xeno-by** [comment]
At the moment (https://github.com/twitter/rsc/commit/a8b3d836b35e9ce5e6b81bd968398fec6ab06cce), this happens to work. I remember doing a bunch of work in this area for other tickets, and apparently it took care of this ticket as well.
